_docs/4-kubernetes/2-helloworld.md

```diff
@@ -0,0 +1,178 @@
+   $ kubectl get routingrule -o json
+   {
+       "apiVersion": "v1",
+       "items": [
+           {
+               "apiVersion": "amalgam8.io/v1",
+               "kind": "RoutingRule",
+               "metadata": {
+                   "creationTimestamp": "2017-02-07T12:45:56Z",
+                   "name": "set-helloworld-default-v1",
+                   "namespace": "default",
+                   "resourceVersion": "1720622",
+                   "selfLink": "/apis/amalgam8.io/v1/namespaces/default/routingrules/set-helloworld-v1-default",
+                   "uid": "5e53a321-ed33-11e6-a49c-fa163efdea7d"
+               },
+               "spec": {
+                   "destination": "helloworld",
+                   "id": "set-helloworld-v1-default",
+                   "priority": 1,
+                   "route": {
+                       "backends": [
+                           {
+                               "tags": [
+                                   "version=v1"
+                               ]
+                           }
+                       ]
+                   }
+               },
+               "status": {
+                   "state": "valid"
+               }
+           }
+       ],
+       "kind": "List",
+       "metadata": {},
+       "resourceVersion": "",
+       "selfLink": ""
+   }
+   ```
+
+1. Confirm that all traffic is being directed to the v1 instance, by running the following curl command multiple times.
+
+   ```bash
+   $ for i in {1..4} do curl http://$GATEWAY_URL/helloworld/hello; done
+   ...
+   ```
+
+   You can see that the traffic is continually routed between the v1 instances only, in a random fashion.
+
+1. Next, we will split traffic between helloworld v2 (25%) and v1 (the remaining 75%)
+
+   Run the following command to send 25% of the traffic to helloworld v2, leaving the rest (75%) on v1:
+
+   ```bash
+   $ kubectl create -f testing/test-scripts/helloworld-v1-v2-route-rules.yaml
+   ```
+
+1. Run this curl command several times:
+
+   ```bash
+   curl http://$GATEWAY_URL/helloworld/hello
+   ```
+
+   You will see alternating responses from all 4 helloworld instances, where approximately 1 out of every 4 (25%) responses
+   will be from a "v2" instances, and the other responses from the "v1" instances:
+
+   ```bash
+   $ for i in {1..8} do curl http://$GATEWAY_URL/helloworld/hello; done
+   ```
+
+   Note: if you use a browser instead of cURL to access the service and continually refresh the page,
+   it will always return the same version (v1 or v2), because a cookie is set to maintain version affinity.
+   However, the browser still alternates in a random manner between instances of the specific version.
+
+### Cleanup
+
+1. Delete the routing rules
+
+   ```bash
+   $ kubectl delete -f testing/test-scripts/helloworld-default-route-rules.yaml
+   routingrule "set-helloworld-v1-default" deleted
+   $ kubectl delete -f testing/test-scripts/helloworld-v1-v2-route-rules.yaml
+   ...
```

Comment:
done. `...` were mostly placeholders for additional output that was blocked on the a8 fixes to k8srules. Will add actual output in follow-up commit.